eva_vcf_merge/detect.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2021 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+from enum import Enum
+
+import pysam
+
+from eva_vcf_merge.utils import are_all_elements_unique
+
+
+class SampleSetType(Enum):
+    SINGLE_SET = 'single set'
+    UNSORTED_SINGLE_SET = 'unsorted single set'
+    UNIQUE_SETS = 'unique sets'
+    OVERLAPPING_SETS = 'overlapping sets'
+
+
+class MergeType(Enum):
+    HORIZONTAL = 'horizontal'
+    VERTICAL = 'vertical'
+
+
+def detect_merge_type(vcf_files):
+    """
+    Detect what type of merge should be used for a list of VCFs.
+
+    :param vcf_files: list of vcf filepaths
+    :return: MergeType or None if it could not be detect
+    """
+    file_to_sample_names = {}
+    # retrieve all the sample_names from the VCF files
+    for file_path in vcf_files:
+        file_to_sample_names[file_path] = get_samples_from_vcf(file_path)
+    # Check that all the samples are the same and in the same order to enable horizontal merging
+    sample_info = compare_sample_sets(file_to_sample_names.values())
+    if sample_info == SampleSetType.UNIQUE_SETS:
+        return MergeType.HORIZONTAL
+    elif sample_info == SampleSetType.SINGLE_SET:
```

Comment:
Did bcftools merge have issues VCFs with UNSORTED_SINGLE_SET i.e., swapped sample order?

Reply:
Do you mean bcftools concat (i.e. "vertical merge")?  The [docs for concat](http://samtools.github.io/bcftools/bcftools.html#concat) call for "the same sample columns appearing in the same order" but to be honest I haven't actually tried it.

Reply:
No worries. Yes, I did mean concat. Even if it works, it is good to not try undocumented things. So I agree with the decision here to bail on such a scenario.